src/containers/Container.js

```diff
@@ -23,23 +24,28 @@ const Container = (props) => (
       </div>
 
       <div className="container">
-
         <div className="row">
-          <div className="col-md-4">
-            <h2>Heading</h2>
-            <p>Donec id elit non mi porta gravida at eget metus. Fusce dapibus, tellus ac cursus commodo, tortor mauris condimentum nibh, ut fermentum massa justo sit amet risus. Etiam porta sem malesuada magna mollis euismod. Donec sed odio dui. </p>
-            <p><a className="btn btn-secondary" href="#" role="button">View details &raquo;</a></p>
-          </div>
-          <div className="col-md-4">
-            <h2>Heading</h2>
-            <p>Donec id elit non mi porta gravida at eget metus. Fusce dapibus, tellus ac cursus commodo, tortor mauris condimentum nibh, ut fermentum massa justo sit amet risus. Etiam porta sem malesuada magna mollis euismod. Donec sed odio dui. </p>
-            <p><a className="btn btn-secondary" href="#" role="button">View details &raquo;</a></p>
-          </div>
-          <div className="col-md-4">
-            <h2>Heading</h2>
-            <p>Donec <Link href="#">sed</Link> odio dui. Cras justo odio, dapibus ac facilisis in, egestas eget quam. Vestibulum id ligula porta felis euismod semper. Fusce dapibus, tellus ac cursus commodo, tortor mauris condimentum nibh, ut fermentum massa justo sit amet risus.</p>
-            <p><a className="btn btn-secondary" href="#" role="button">View details &raquo;</a></p>
-          </div>
+          {
+            [1,2,3,4,5,6,7,8].map((i) => {
+              return (
+                <div className="col-md-4" key={i + "row"} p={i+"row"}>
+                  <div className="card mb-4 box-shadow">
+                  <Photo className="card-img-top" src="2707f16843bff61726e269cb195bede8713764b52421f6d7111425c130a187c9.JPG"  style="height: 225px; width: 100%; display: block;" />
+                    <div className="card-body">
+                      <p className="card-text">This is a wider card with supporting text below as a natural lead-in to additional content. This content is a little bit longer.</p>
+                      <div className="d-flex justify-content-between align-items-center">
+                        <div className="btn-group">
+                          <button type="button" className="btn btn-sm btn-outline-secondary">View</button>
+                          <button type="button" className="btn btn-sm btn-outline-secondary">Edit</button>
+                        </div>
+                        <small className="text-muted">9 mins</small>
+                      </div>
+                    </div>
+                  </div>
+                </div>
```

Comment:
Is it pretty standard to have the actual html in js code like this? Feels like this could be separated out into an html flie, but not sure how familiar you are with doing that (I defintely don't know how).

Reply:
Yeah. so this is literally how react works. You define your small components into a single JS file.

a react component, and that component has a `render()` function. That function holds the HTML and gets called whenever something in the component's changes. 


React components can also have states which are semi-permanent storage held in memory until the component is released (for instance reloading the page, or even switching pages when the component is no longer rendered).

```
import react from 'react'
class Greeting extends React.component {
   constructor(props) {
     super(props);
     this.state = { hello: 'world' };
   }
  
  render() {
    return (
      <div>{this.state.hello}</div>
    );
  }
}
export default Greeting;
```

This can also be shortcut into a more compact syntax that has other benefits i don't remember:

```
import react from 'react'
  const [hello, setHello] = useState('world');
  
  return (
    <div>{this.state.hello}</div>
  );
};
```